### PR TITLE
feat(api): appeal-case data

### PR DESCRIPTION
**Pull request by beejjacobs**
## Ticket Number

https://pins-ds.atlassian.net/browse/AAPD-954

## Description of change

- Added in fields to AppealCase from the Data Model
- Created a /v2/appeal-cases endpoint
- fixed integration tests (failures weren't reported)
- updated Jest to support subpath imports
- to do: tests for this feature!

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [ ] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/appeals-service-api/src/db/seed/util.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ */
+function dateFromYMD(year, month, day) {
+	const yearStr = year.toString();
+	month++; // one-index
+	// wrap around
+	if (month < 1) {
+		month += 12;
+	} else if (month > 12) {
+		month -= 12;
+	}
```
Comment (Si3rr4wow) [inline packages/appeals-service-api/src/db/seed/util.js:19]: `Date.setMonth` expects zero index and wraps ;)

File: packages/appeals-service-api/src/routes/v2/appeal-cases/controller.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+/**
+ * @type {import('express').RequestHandler}
+ */
+async function getCount(req, res, next) {
+	if ('lpa-code' in req.query) {
+		countByLpaCode(req, res, next);
+		return;
+	}
+	throw ApiError.withMessage(400, 'lpa-code is required');
+}
```
Comment (Si3rr4wow) [inline packages/appeals-service-api/src/routes/v2/appeal-cases/controller.js:90]: Maybe invert this so the error is thrown by a guard clause?

File: packages/appeals-service-api/src/routes/v2/appeal-cases/repo.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	 * @param {string} caseReference
+	 * @returns {Promise<AppealCase|null>}
+	 */
+	async getByCaseReference(caseReference) {
+		return await dbClient.appealCase.findUnique({
+			where: {
+				caseReference
+			}
+		});
+	}
```
Comment (Si3rr4wow) [inline packages/appeals-service-api/src/routes/v2/appeal-cases/repo.js:23]: is async await necessary here if the goal is to return a promise?

File: packages/appeals-service-api/src/routes/v2/appeal-cases/controller.js (hunk 2 above)
Comment (beejjacobs) [reply packages/appeals-service-api/src/routes/v2/appeal-cases/controller.js:90]: I was using the same pattern as `list`, though it does look odd with just one option in there.

File: packages/appeals-service-api/src/routes/v2/appeal-cases/spec.yaml [hunk 4]
```diff
@@ -0,0 +1,94 @@
+paths:
```
Comment (bryn-pins) [inline packages/appeals-service-api/src/routes/v2/appeal-cases/spec.yaml:1]: thoughts on using tags, little annoying needing to add to main spec and here, but it looks nicer to me

**Review by bryn-pins — COMMENTED** [review]
Apart from tests, think it looks good

I don't think we need to require tests on this PR as it was already missed, can do separately if needed

File: packages/appeals-service-api/src/routes/v2/appeal-cases/spec.yaml (hunk 4 above)
Comment (beejjacobs) [reply packages/appeals-service-api/src/routes/v2/appeal-cases/spec.yaml:1]: I hadn't appreciated what they did in the UI. added in.

You don't have to add to the root, but if you do you can control order in the UI which is handy.

**Comment by beejjacobs** [comment]
This work is required by other tickets going forward, so going to merge as-is and add tests after.

**Review by bryn-pins — APPROVED** [review]

**Pull request merged**